apiserver/facades/client/bundle/bundle.go

```diff
@@ -622,10 +621,19 @@ func (b *BundleAPI) endpointBindings(bindings map[string]string) (map[string]str
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	names := set.NewStrings()
 	for k, v := range endpointsWithSpaceNames {
-		if v == network.DefaultSpaceName {
+		if v == "" {
 			delete(endpointsWithSpaceNames, k)
 		}
+		names.Add(v)
+	}
+	// Assumption: if all endpoints are in the same space,
```

Comment:
I think this assumption is wrong. Case in point: 
```
juju deploy keystone --bind "top-secret"
```

In this scenario the operator is definitely using spaces; it's just that he selected a **non-alpha** space for everything. I think a more accurate predicate for the comparison below would also include a `names.Values()[0] != network.AlphaSpaceName`.

Reply:
This is changing to "if all EP space in the bundle are the same".  Discussion during standup.  We're moving away from excluding the AlphaSpaceName with the paradigm shift in spaces to having all "real" spaces.